ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -399,8 +399,11 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
       }
     }
 
-    final Account sender = updater.get(senderAddress);
-    final long nonce = sender != null ? sender.getNonce() : 0L;
+    final long nonce =
+        callParams
+            .getNonce()
+            .orElse(
```

Comment:
For efficiency, should not be `orElseGet` to only compute it is the nonce is not present in the call params?

Reply:
> For efficiency, should not be `orElseGet` to only compute it is the nonce is not present in the call params?

@fab-10 https://github.com/hyperledger/besu/pull/8145